src/Context.ts

```diff
@@ -59,3 +65,10 @@ export type SubscriptionContext = {
   log: FastifyBaseLogger;
   userId?: string;
 };
+
+export type BaseContext = Omit<Context, 'userId' | 'trackingId'>;
+
+export type AuthContext = BaseContext & {
+  userId: string;
+  trackingId: string;
+};
```

Comment:
Two new contexts:
- `BaseContext` - used as type base in schema resolver
- `AuthContext` used in query & mutations with `@auth` directive to make `userId` being defined which is guaranteed by the directive